File: db-services/db-palette-service/src/main/java/com/qreal/wmp/db/palette/model/PaletteView.java
package com.qreal.wmp.db.palette.model;

import com.qreal.wmp.thrift.gen.TPaletteView;
import lombok.Data;

/* View of palette for client side.*/
@Data
public class PaletteView
{
    private Long id;

    private String name;

    public PaletteView(Long id, String name) {
        this.id = id;
        this.name = name;
    }

    /** Constructor-converter from Thrift TPalette to Palette.*/
    public PaletteView(TPaletteView tPalette) {
        if (tPalette.isSetId()) {
            id = tPalette.getId();
        }

        if (tPalette.isSetName()) {
            name = tPalette.getName();
        }
    }

    /** Converter from Palette to Thrift TPalette.*/
    public TPaletteView toTPalette() {
        TPaletteView tPalette = new TPaletteView();

        if (id != null) {
            tPalette.setId(id);
        }

        if (name != null) {
            tPalette.setName(name);
        }

        return tPalette;
    }
}

File: editor-service/src/main/webapp/app/dsm/diagram/model/PaletteView.ts
export class PaletteView {

    private id: number;
    private name: string;

    public static createFromDAO(palette: TPaletteView) {
        return new PaletteView(palette.id, palette.name)
    }

    constructor(id: number, name: string) {
        this.id = id;
        this.name = name;
    }

    public setId(id: number): void {
        this.id = id;
    }

    public getId(): number {
        return this.id;
    }

    public setName(name: string): void {
        this.name = name;
    }

    public getName(): string {
        return this.name;
    }

}